wavefront/wavefront.go

```diff
@@ -91,8 +91,10 @@ func VerboseLogging() Option {
 // Exporter is the main exporter
 type Exporter struct {
 	sender senders.Sender
-	sem    chan struct{}
-	wg     sync.WaitGroup
+
+	wg      sync.WaitGroup
+	lock    sync.Mutex
```

Comment:
You may want to consider making this a `sync.RWMutex` since it's only written to once when the exporter stops.